aten/src/ATen/native/HalfFloatOps.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include <ATen/ATen.h>
+
+#include "fbgemm/Fbgemm.h"
+#include <fbgemm/FbgemmConvert.h>
+
+namespace at {
+namespace native {
+
+namespace {
+
+inline void FloatToFloat16_ref(
+    const float* in,
+    at::Half* out,
+    size_t N,
+    bool do_clip = false) {
+  if (do_clip) {
+    constexpr float FP16_MAX = 65504.f;
+    for (size_t i = 0; i < N; ++i) {
+      out[i] = std::max(-FP16_MAX, std::min(in[i], FP16_MAX));
+    }
+  } else {
+    for (size_t i = 0; i < N; ++i) {
+      out[i] = in[i];
+    }
+  }
+}
+
+inline void Float16ToFloat_ref(const at::Half* in, float* out, size_t N) {
+  for (size_t i = 0; i < N; ++i) {
+    out[i] = in[i];
+  }
+}
+
+} // namespace
+
+
+Tensor half_to_float(const Tensor& input) {
```

Comment:
Some documentation on these conversion ops wil be useful. How is this different from the existing `.to(dtype)` in pytorch?